### PR TITLE
Don't offer to add a using for the namespace that the code is already contained within

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/11608
Fixes https://github.com/dotnet/roslyn/issues/9597

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

**Comment by balajikris** [comment]
:+1:

**Comment by CyrusNajmabadi** [comment]
Retest windows_eta_open_prtest please

**Comment by CyrusNajmabadi** [comment]
ETA test needs to be updated in closed.

File: src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs [hunk 1]
```diff
@@ -44,8 +47,26 @@ private class SymbolReferenceFinder
...
                 _containingTypeOrAssembly = _containingType ?? (ISymbol)semanticModel.Compilation.Assembly;
-                _namespacesInScope = owner.GetNamespacesInScope(semanticModel, node, cancellationToken);
                 _syntaxFacts = document.Project.LanguageServices.GetService<ISyntaxFactsService>();
+
+                _namespacesInScope = GetNamespacesInScope(cancellationToken);
+            }
+
+            private ISet<INamespaceSymbol> GetNamespacesInScope(CancellationToken cancellationToken)
+            {
+                // Add all hte namespaces brought in by imports/usings.
```
Comment (Pilchie) [inline src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs]: Speelnig

File: src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs [hunk 2]
```diff
@@ -313,7 +313,7 @@ private static IList<ParameterName> GenerateNames(IList<string> reservedNames, L
         public static ISet<INamespaceSymbol> GetUsingNamespacesInScope(this SemanticModel semanticModel, SyntaxNode location)
         {
             // Avoiding linq here for perf reasons. This is used heavily in the AddImport service
-            HashSet<INamespaceSymbol> result = null;
+            var result = new HashSet<INamespaceSymbol>();
```
Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs]: Oh.  Hmm.  I suspect we might run into problems with merged namespaces, if I declare myself inside `namespace System`, but also have references to assemblies which define `System`.  Do we need a different kind of equality here?

Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs]: Will try to add tests for that case, but i think we should be fine.

**Comment by CyrusNajmabadi** [comment]
retest windows_eta_open_prtest please

**Comment by CyrusNajmabadi** [comment]
@Pilchie I made it so we always store 'Compilation-Namespaces' so that we're always operating on the same namespace currency.

File: src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs [hunk 3]
```diff
@@ -44,8 +47,31 @@ private class SymbolReferenceFinder
...
+                var set = _owner.GetImportNamespacesInScope(_semanticModel, _node, cancellationToken);
```
Comment (Pilchie) [inline src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs]: Namespaces coming back from this call will still be MergedNamespace symbols, not compilation ones AFAICT.

Comment (Pilchie) [inline src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs]: Ah, I misunderstood what was meant by compilation namespaces in this case.  I was thinking it was a namespace symbol within the compilation's assembly, but it *is* the merged namespaces across all assemblies in the namespace.  I withdraw my concern.

**Comment by Pilchie** [comment]
Still a spelling nit, but :+1:

**Pull request merged**